packages/react-jss/src/withStyles.js

```diff
@@ -44,23 +35,24 @@ const createWithStyles: CreateWithStyles = <Theme>(styles, options = {}) => {
         classesProp ? mergeClasses(sheetClasses, classesProp) : sheetClasses
     )
 
+    const useStyles = createUseStyles(
+      (styles: any),
+      ({
+        theming,
+        index,
+        name: displayName,
+        ...sheetOptions
+      }: any)
+    )
+
     const WithStyles = React.forwardRef((props: HOCProps<Theme, Props>, ref) => {
       const theme = React.useContext(ThemeContext)
 
-      const useStyles = React.useMemo(
-        () =>
-          createUseStyles(styles, {
-            theming,
-            index,
-            name: displayName,
-            ...sheetOptions
-          }),
-        [styles, theming, index, displayName, sheetOptions]
-      )
-
       const newProps = {...props}
 
-      if (injectTheme && newProps.theme == null) newProps.theme = theme
+      if (injectTheme && newProps.theme == null) {
+        ;(newProps: any).theme = theme
```

Comment:
- any
- I think we can avoid needing semicolon first

Reply:
Yes, but semicolon added by prettier and I don't know way

Reply:
I fixed